pkg/netceptor/packetconn.go

```diff
@@ -42,35 +79,38 @@ func (s *Netceptor) ListenPacket(service string) (*PacketConn, error) {
 		return nil, fmt.Errorf("service %s is already listening", service)
 	}
 	_ = s.addNameHash(service)
-	pc := &PacketConn{
-		s:            s,
-		localService: service,
-		recvChan:     make(chan *MessageData),
-		advertise:    false,
-		adTags:       nil,
-		connType:     ConnTypeDatagram,
-		hopsToLive:   s.maxForwardingHops,
+	pc, err := NewPacketConn(s, service, ConnTypeDatagram)
+	if err != nil {
+		return nil, err
 	}
-	pc.startUnreachable()
-	s.listenerRegistry[service] = pc
 
 	return pc, nil
 }
 
 // ListenPacketAndAdvertise returns a datagram listener, and also broadcasts service
 // advertisements to the Receptor network as long as the listener remains open.
-func (s *Netceptor) ListenPacketAndAdvertise(service string, tags map[string]string) (*PacketConn, error) {
-	pc, err := s.ListenPacket(service)
+func (s *Netceptor) ListenPacketAndAdvertise(service string, tags map[string]string) (PacketConner, error) {
```

Comment:
It seems `ListenPacket` does a lot more error checking than this function, why? 

As the checks not valid between the two functions or were they missed?